dhcp-controller/controllers/dhcpserver_controller.go

```diff
@@ -0,0 +1,202 @@
+	//_ = log.FromContext(ctx)
+
+	//log := r.Log.WithValues("dhcpservers", req.NamespacedName)
+
+	var dhcpConfigReq = dhcpv1alpha1.DHCPServerList{}
+	if err := r.List(ctx, &dhcpConfigReq); err != nil {
+		log.Error(err, "unable to fetch DHCPServer")
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	log.Info("Reconciling")
+
+	serverList := dhcpConfigReq.Items
+	if len(serverList) > 0 {
+		for _, server := range serverList {
+			log.Infof("server %s", server.Name)
+			result, err := r.ensureServer(req, server, r.backendDeployment(server))
+			if result != nil {
+				log.Error(err, "DHCP server Not ready")
+				return *result, err
+			}
+		}
+	} else {
+		log.Info("serverList is Empty")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *DHCPServerReconciler) ensureServer(request reconcile.Request,
+	server dhcpv1alpha1.DHCPServer,
+	dep *appsv1.Deployment,
+) (*reconcile.Result, error) {
+
+	// See if deployment already exists and create if it doesn't
+	found := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Name:      dep.Name,
+		Namespace: server.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		// Create the deployment
+		err = r.Create(context.TODO(), dep)
+
+		if err != nil {
+			// Deployment failed
+			return &reconcile.Result{}, err
+		} else {
+			// Deployment was successful
+			return nil, nil
+		}
+	} else if err != nil {
+		// Error that isn't due to the deployment not existing
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+// backendDeployment is a code for Creating Deployment
+func (r *DHCPServerReconciler) backendDeployment(v dhcpv1alpha1.DHCPServer) *appsv1.Deployment {
+
+	size := int32(1)
+	memReq := resource.NewQuantity(64*1024*1024, resource.BinarySI)
+	memLimit := resource.NewQuantity(128*1024*1024, resource.BinarySI)
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.Name,
+			Namespace: v.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &size,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": "dnsmasq", "name": v.Name},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      map[string]string{"app": "dnsmasq", "name": v.Name},
+					Annotations: map[string]string{"k8s.v1.cni.cncf.io/networks": v.Spec.NetworkName},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{{
+						Image: "docker.io/ataa/dnsmasq:latest",
+						SecurityContext: &corev1.SecurityContext{
+							Capabilities: &corev1.Capabilities{
+								Add: []corev1.Capability{
+									"NET_ADMIN",
+								}}},
+						Resources: corev1.ResourceRequirements{
+							Limits: corev1.ResourceList{
+								"memory": *memLimit,
+							},
+							Requests: corev1.ResourceList{
+								"memory": *memReq,
+							},
+						},
+						ImagePullPolicy: corev1.PullAlways,
```

Comment:
image pull policy should be IfNotPresent for airgap. And need to add this to the image bundle list in airctl.

Also can we not use latest dnsmasq? Let's use most recent stable version

Reply:
For the second part, the dnsmsaq image is what we build in [dhcpserver](https://github.com/platform9/luigi/tree/private/ashutosh/dhcp-controller/dhcp-controller/dhcpserver). Will be tagged appropriately and uploaded once we finish changes